src/foam/dashboard/view/DashboardView.js

```diff
@@ -98,29 +128,69 @@ foam.CLASS({
           'grid-gap': this.gap$
         });
 
-      Object.keys(this.widgets).map(async menuId => {
-        let menu = await this.menuDAO.find(menuId);
-        if ( menu ) {
-          let aspectRatio = this.widgets[menu.id];
-          widgetContainer.startContext().start(menu.handler.view).style({
-            'grid-column': this.containerWidth$.map(v => {
-              return 'span ' + (v?.cols ? Math.min(aspectRatio.split('/')[0], v?.cols) : aspectRatio.split('/')[0]);
-            }),
-            'grid-row': 'span ' + aspectRatio.split('/')[1]
-          }).end();
-        }
-      });
-
       this
-        .addClass(this.myClass())
-        .enableClass(this.myClass('main'), this.main)
-        .start()
-          .hide(!this.dashboardTitle)
-          .enableClass('h500', this.dashboardTitle)
-          .style({ height: '2em' })
-          .add(this.dashboardTitle)
-        .end()
-        .tag(widgetContainer)
+      .addClass(this.myClass())
+      .enableClass(this.myClass('main'), this.main)
+      .start()
+        .hide(!this.dashboardTitle)
+        .enableClass('h500', this.dashboardTitle)
+        .style({ height: '2em' })
+        .add(this.dashboardTitle)
+      .end()
+      .tag(widgetContainer)
+
+      let menuLength = Object.keys(this.widgets).length
```

Comment:
This would be more efficient if you did:
let menuArray = Object.keys(this.widgets);
let menuLength = menuArray.length;

Reply:
How? It looks the exact same.